Update API docs and getting started guide (#836)
hashicorp/terraform-website: content/source/docs/enterprise/api/policy-sets.html.md

[Sentinel Policy as Code](../sentinel/index.html) is an embedded policy as code framework integrated with Terraform Enterprise.

> Policy sets are groups of policies that are applied together to related workspaces. By using policy sets, you can group your policies by attributes such as environment or region. Individual policies within a policy set will only be checked for workspaces that the policy set is attached to. Policy sets can group indidual policies created via the [policies API](./policies.html), or act as versioned sets which are sourced from either a version control system (such as GitHub) or uploaded as a whole via the [policy set versions API](#create-a-policy-set-version).

Reviewer: ```suggestion
Policy sets are groups of policies that are applied together to related workspaces. By using policy sets, you can group your policies by attributes such as environment or region. Individual policies within a policy set will only be checked for workspaces that the policy set is attached to. Policy sets can group indidual policies created via the [policies API](./policies.html), or act as versioned sets which are either sourced from a version control system (such as GitHub) or uploaded as a whole via the [policy set versions API](#create-a-policy-set-version).
```

`data.attributes.vcs-repo`                    | object          | `null`  | VCS repository information. When present, the policies and configuration will be sourced from the specified VCS repository instead of being defined within TFE. Note that this option and `policies` relationships are mutually exclusive and may not be used simultaneously.
`data.attributes.vcs-repo.branch`             | string          | `null`  | The branch of the VCS repo. If empty, the VCS provider's default branch value will be used.
`data.attributes.vcs-repo.identifier`         | string          |         | The identifier of the VCS repository in the format `<namespace>/<repo>`. For example, on GitHub, this would be something like `hashicorp/my-policy-set`.
`data.attributes.vcs-repo.oauth-token-id`     | string          |         | The OAuth Token ID to use to connect to the VCS host.
`data.attributes.vcs-repo.ingress-submodules` | boolean         | `false` | Determines whether repository submodules will be instantiated during the clone operation.

> `data.attributes.policies-path`               | string          | `null`  | The sub-path within the attached VCS repository to clone. All files and directories outside of this sub-path will be ignored. This option may only be specified when a VCS repo is present. **Note:** When this option is configured, commits in the repository which are outside of the given policies-path will be ignored.

Reviewer: Hmm. This should be clearer, but let's see whether I can actually manage that:

```suggestion
`data.attributes.policies-path`               | string          | `null`  | The subdirectory of the attached VCS repository that contains the policies for this policy set. Files and directories outside of this sub-path will be ignored, and changes to those unrelated files won't cause the policy set to be updated. This option may only be specified when a VCS repo is present.
```

> `data.relationships.policies.data[]`          | array\[object\] | `[]`    | A list of resource identifier objects that defines which policies will be members of the new set. These objects must contain `id` and `type` properties, and the `type` property must be `policies` (e.g. `{ "id": "pol-u3S5p2Uwk21keu1s", "type": "policies" }`). **Note:** This option will be deprecated in the future in favor of VCS policy sets.

Reviewer: ```suggestion
`data.relationships.policies.data[]`          | array\[object\] | `[]`    | **DEPRECATED.** A list of resource identifier objects that defines which policies will be members of the new set. These objects must contain `id` and `type` properties, and the `type` property must be `policies` (e.g. `{ "id": "pol-u3S5p2Uwk21keu1s", "type": "policies" }`). **Important:** This deprecated option will be removed in the future in favor of VCS policy sets.
```

}
```

### Sample Payload with VCS repository

```json
{
  "data": {
    "attributes": {
      "name": "production",
      "description": "This set contains policies that should be checked on all production infrastructure workspaces.",
      "global": false,
      "policies-path": "/policy-sets/foo",
      "vcs-repo": {
        "branch": "master",
        "identifier": "hashicorp/my-policy-sets",
        "ingress-submodules": false,
        "oauth-token-id": "ot-7Fr9d83jWsi8u23A"
      }
    },
    "relationships": {
      "workspaces": {
        "data": [
          { "id": "ws-2HRvNs49EWPjDqT1", "type": "workspaces" }
        ]
      }
    },
    "type": "policy-sets"
  }
}
```

> 

Reviewer 2: Consider inverting this? "Sample Payload", and then take the existing one and put it below it as "Sample Payload (Deprecated)" or similar.

Reviewer 2: Same for the other examples.

[200][] | [JSON API document][] (`type: "policy-sets"`) | Request was successful
[404][] | [JSON API error object][]                     | Organization not found, or user unauthorized to perform action

### Query Parameters

This endpoint supports pagination [with standard URL query parameters](./index.html#query-parameters); remember to percent-encode `[` as `%5B` and `]` as `%5D` if your tooling doesn't automatically encode URLs.

Parameter           | Description
--------------------|------------

> `filter[versioned]` | **Optional.** Allows filtering policy sets based on whether they are versioned (VCS or API), or manually updated. Accepts a boolean true/false value. If omitted, all policy sets are returned.

Reviewer: ```suggestion
`filter[versioned]` | **Optional.** `filter[true]` will only show versioned policy sets (VCS-managed or API-managed), and `filter[false]` will only show manually updated policy sets. If omitted, all policy sets are returned.
```

Author: The `versioned` key still needs to be specified so like `filter[versioned]=true` or `filter[versioned]=false`. I hadn't seen this long-form used anywhere else, but I'm not opposed to it.

> `include`           | **Optional.** Allows including related resource data. Value must be one of `workspaces`, `policies`, or `current_version`.

Reviewer 2: I think it's good you're doc'ing this `include` stuff as we're really bad at it in general, but where we do doc it we usually leave it as a separate section, [like in the Workspaces API](https://www.terraform.io/docs/enterprise/api/workspaces.html#available-related-resources). Up to @nfagerlund really, I just thought I'd point out where we're differing from the consistent norm here.

Reviewer 2: (though if we're documenting the pagination stuff here I see no reason not to also...include....`include`, too) 🤷‍♂

Author: I had also noticed that, and thought "this is probably where I'd come looking for this option", so thus I included it here.

`data.attributes.vcs-repo`                    | object  | (previous value) | VCS repository information. When present, the policies and configuration will be sourced from the specified VCS repository instead of being defined within TFE. Note that this option and `policies` relationships are mutually exclusive and may not be used simultaneously.
`data.attributes.vcs-repo.branch`             | string  | (previous value) | The branch of the VCS repo. If empty, the VCS provider's default branch value will be used.
`data.attributes.vcs-repo.identifier`         | string  | (previous value) | The identifier of the VCS repository in the format `<namespace>/<repo>`. For example, on GitHub, this would be something like `hashicorp/my-policy-set`.
`data.attributes.vcs-repo.oauth-token-id`     | string  | (previous value) | The OAuth Token ID to use to connect to the VCS host.
`data.attributes.vcs-repo.ingress-submodules` | boolean | (previous value) | Determines whether repository submodules will be instantiated during the clone operation.

> `data.attributes.policies-path`               | boolean | (previous value) | The sub-path within the attached VCS repository to clone. All files and directories outside of this sub-path will be ignored. This option may only be specified when a VCS repo is present. **Note:** When this option is configured, commits in the repository which are outside of the given policies-path will be ignored. When this option is changed, the repository will be automatically cloned again to ensure this path value is reflected in the policy set contents.

Reviewer: (same as above)

  https://app.terraform.io/api/v2/policy-sets/polset-3yVQZvHzf5j3WRJ1
```

## Create a Policy Set Version

> For versioned policy sets which have no VCS repository attached, versions of policy code may be uploaded directly to the API by creating a new Policy Set Version, and uploading a tarball (tar.gz) of data to it.

Reviewer: ```suggestion
For versioned policy sets which have no VCS repository attached, versions of policy code may be uploaded directly to the API by creating a new policy set version and, in a subsequent request, uploading a tarball (tar.gz) of data to it.
```
